examples/medplum-provider/src/pages/encounter/EncounterModal.tsx

```diff
@@ -0,0 +1,187 @@
+import { IconCircleCheck, IconCircleOff } from '@tabler/icons-react';
+import { createReference, normalizeErrorString } from '@medplum/core';
+import { Coding, Encounter, ValueSetExpansionContains } from '@medplum/fhirtypes';
+import { usePatient } from '../../hooks/usePatient';
+
+export const EncounterModal = (): JSX.Element => {
+  const navigate = useNavigate();
+  const medplum = useMedplum();
+  const location = useLocation();
+  const patient = usePatient();
+  const [types, setTypes] = useState<ValueSetExpansionContains[]>([]);
+  const [encounterClass, setEncounterClass] = useState<Coding | undefined>();
+  // Todo: create a resusable type
+  const [status, setStatus] = useState<
+    | 'planned'
+    | 'arrived'
+    | 'triaged'
+    | 'in-progress'
+    | 'onleave'
+    | 'finished'
+    | 'cancelled'
+    | 'entered-in-error'
+    | 'unknown'
+  >('planned');
+  const isOpen = location.pathname.endsWith('/Encounter/new');
+
+  const handleClose = (): void => {
+    navigate(-1);
+  };
+
+  const handleCreateEncounter = (): void => {
+    if (!patient || !encounterClass) {
+      return;
+    }
+
+    const encounterData: Encounter = {
+      resourceType: 'Encounter',
+      status: status,
+      statusHistory: [],
+      class: encounterClass,
+      classHistory: [],
+      type: [
+        {
+          coding: types,
+        },
+      ],
+      subject: createReference(patient),
+    };
+
+    medplum
+      .createResource(encounterData)
+      .then((encounter) => {
+        showNotification({
+          icon: <IconCircleCheck />,
+          title: 'Success',
+          message: 'Encounter created',
+        });
+        console.log('Encounter created:', encounter);
+        // Redirect to actual chart
+        navigate(`/Patient/6d8f3e99-f763-422c-946d-68c95e394a82/Encounter/chart`);
+      })
+      .catch((err) => {
+        showNotification({
+          color: 'red',
+          icon: <IconCircleOff />,
+          title: 'Error',
+          message: normalizeErrorString(err),
+        });
+      });
+  };
+
+  return (
+    <Modal
+      opened={isOpen}
+      onClose={handleClose}
+      size="xl"
+      title="New encounter"
+      styles={{
+        title: {
+          fontSize: '1.125rem',
+          fontWeight: 600,
+        },
+      }}
+    >
+      <div style={{ display: 'grid', gridTemplateColumns: '1fr 1fr', gap: '2rem' }}>
+        <div style={{ display: 'flex', flexDirection: 'column', gap: '1.5rem' }}>
+          {/* TODO: Enable change of Patient */}
+          <ResourceInput resourceType="Patient" name="Patient-id" defaultValue={patient} disabled={true} />
+
+          <ValueSetAutocomplete
+            name="type"
+            label="Type"
+            binding="http://hl7.org/fhir/ValueSet/service-type"
+            withHelpText={true}
+            onChange={(items: ValueSetExpansionContains[]) => {
+              console.log(items);
+              setTypes(items);
+            }}
```

Comment:
1. Remove `console.log`
2. For single function call, please use inline

i.e.,

```ts
onChange={setTypes}
```